astropy/coordinates/builtin_frames/utils.py

```diff
@@ -236,25 +279,61 @@ def atciqz(rc, dc, astrom):
     same approach used by the ERFA functions above, except that they use a threshold of
     9 arcseconds.
 
+    In addition, ERFA's atciqz assumes a distant source, so there is no difference between
+    the object-Sun vector and the observer-Sun vector. This can lead to errors of up to a
+    few arcseconds in the worst case (e.g a Venus transit).
+
     Parameters
     ----------
-    rc : float or `~numpy.ndarray`
-        right ascension, radians
-    dc : float or `~numpy.ndarray`
-        declination, radians
+    astrometric : `~astropy.coordinates.CartesianRepresentation`
+        Astrometric ICRS position of object from observer
     astrom : eraASTROM array
         ERFA astrometry context, as produced by, e.g. ``eraApci13`` or ``eraApcs13``
 
     Returns
     --------
     ri : float or `~numpy.ndarray`
+        Right Ascension in radians
     di : float or `~numpy.ndarray`
+        Declination in radians
     """
+    # first of all get distance from object to observer
+    object_observer_distance = astrometric.norm()
+    # ignore parallax effects if no distance, or far away
+    ignore_distance = object_observer_distance.unit == u.one
+
     # BCRS coordinate direction (unit vector).
-    pco = erfa.s2c(rc, dc)
+    if not ignore_distance and u.allclose(astrometric.xyz, 0*u.km):
+        # special casing for positions at origin of frame
+        # replicating the behaviour of erfa.s2c with 0, 0 input
+        # which for reasons unknown gives the vector (1, 0, 0)
+        xhat = astrometric.unit_vectors()['x']
```

Comment:
This seems a bit long. Couldn't one do `tmp = astrometric + CartesianRepresentation([1., 0., 0.], unit=astrometric.x.unit)`? Or just `pco = CartesianRepresentation([1., 0., 0.], unit=u.dimensionless_unscaled)`?